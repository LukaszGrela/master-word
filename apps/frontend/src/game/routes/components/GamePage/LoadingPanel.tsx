import { FC } from 'react';
import { useLanguage } from '../../../../i18n';
import Spinner from '../../../Spinner/Spinner';
import { useGameContext } from '../../../context';

export const LoadingPanel: FC = () => {
  const { getUIText: t } = useLanguage();
  const { loading } = useGameContext();

  return loading ? (
    <h2 className="loading panel">
      <Spinner />
      <span>{t('loading')}</span>
    </h2>
  ) : null;
};
