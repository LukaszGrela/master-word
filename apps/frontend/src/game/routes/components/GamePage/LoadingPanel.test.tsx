import { describe, expect, it, vi } from 'vitest';
import {
  contextRenderer,
} from '../../../context/__tests__/contextRenderer';
import { LoadingPanel } from './LoadingPanel';
import { MockContext } from '../../../../i18n/LanguageContext/__tests__/contextRenderer';

describe('game', () => {
  describe('routes', () => {
    describe('components', () => {
      describe('GamePage', () => {
        describe('LoadingPanel', () => {
          it('renders loading off', () => {
            const { container } = contextRenderer(
              <MockContext
                context={{
                  loading: false,
                  getUIText: vi.fn().mockImplementation(() => 'LOADING'),
                }}
              >
                <LoadingPanel />
              </MockContext>,
              {
                context: {
                  loading: false,
                },
              },
            );

            expect(container).toMatchSnapshot();
          });
          it('renders loading on', () => {
            const { container } = contextRenderer(
              <MockContext
                context={{
                  loading: false,
                  getUIText: vi.fn().mockImplementation(() => 'LOADING'),
                }}
              >
                <LoadingPanel />
              </MockContext>,
              {
                context: {
                  loading: true,
                },
              },
            );

            expect(container).toMatchSnapshot();
          });
        });
      });
    });
  });
});
