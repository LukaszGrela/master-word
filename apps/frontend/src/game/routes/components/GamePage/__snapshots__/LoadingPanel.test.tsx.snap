// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > routes > components > GamePage > LoadingPanel > renders loading off 1`] = `<div />`;

exports[`game > routes > components > GamePage > LoadingPanel > renders loading on 1`] = `
<div>
  <h2
    class="loading panel"
  >
    <div
      class="Spinner"
    >
      <svg
        class="Spinner_svg"
        height="40"
        version="1"
        viewBox="0 0 40 40"
        width="40"
      >
        <circle
          cx="20"
          cy="20"
          id="bg"
          r="18"
        />
        <circle
          cx="20"
          cy="20"
          id="fg"
          r="18"
        />
      </svg>
    </div>
    <span>
      LOADING
    </span>
  </h2>
</div>
`;
